config/src/test/java/org/springframework/security/config/annotation/web/configuration/BaseWebConfig.java

```diff
@@ -15,26 +15,25 @@
  */
 package org.springframework.security.config.annotation.web.configuration;
 
-import org.springframework.context.annotation.Configuration
-import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
-import org.springframework.security.core.userdetails.PasswordEncodedUser
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.core.userdetails.PasswordEncodedUser;
 
 /**
- *
  * @author Rob Winch
+ * @author Konstantin Volivach
  */
 @EnableWebSecurity
 public abstract class BaseWebConfig extends WebSecurityConfigurerAdapter {
-	BaseWebConfig(boolean disableDefaults) {
-		super(disableDefaults)
+
+	public BaseWebConfig(boolean disableDefaults) {
```

Comment:
Ideally, we wouldn't add this class. Would you instead be able to add to the test files something like:

```java
@Configuration
static class UserDetailsServiceConfig {
	@Bean
	public UserDetailsService userDetailsService() {
		return new InMemoryUserDetailsManager(PasswordEncodedUser.user());
	}
}
```

And then use `SpringTestRule` in the tests to instead do:

```java
this.spring.register(MyTestConfig.class, UserDetailsServiceConfig.class).autowire();
```

Reply:
Ok, I made as you say